src/Tests/Tests/Test_Vibrato.cpp

```diff
@@ -0,0 +1,296 @@
+        {
+            delete[] ppfAudioData[i];
+            delete[] ppfOutputData[i];
+            
+            delete[] ppfTempIn[i];
+            delete[] ppfTempOut[i];
+        }
+        delete[] ppfAudioData;
+        delete[] ppfOutputData;
+        
+        delete[] ppfTempIn;
+        delete[] ppfTempOut;
+        
+        CVibrato::destroy(pVibrato);
+    }
+    
+    void testProcess(float** ppfOutputData, int iBlockSize, int iStartSample = 0)
+    {
+        for (int i = 0; i < iNumChannels; i++)
+        {
+            for (int j = 0; j < iBlockSize; j++)
+            {
+                ppfTempIn[i][j] = ppfAudioData[i][iStartSample + j];
+            }
+        }
+
+        pVibrato->process(ppfTempIn, ppfTempOut, iBlockSize);
+
+        for (int i = 0; i < iNumChannels; i++)
+        {
+            for (int j = 0; j < iBlockSize; j++)
+            {
+                ppfOutputData[i][iStartSample + j] = ppfTempOut[i][j];
+            }
+        }
+    }
+};
+
+// Output equals delayed input when modulation amplitude is 0.
+TEST_F(CTestVibrato, ModAmpZero)
+{
+    // e.g., you can use the "VibratoData" contents
+    pVibrato -> reset();
+
+    fModWidth = 0;
+
+    pVibrato -> CVibrato::init(fDelay, fSampleRate, fModWidth, fModFreq, iNumChannels);
+
+    testProcess(ppfOutputData, iBlockSize);
+
+    int iDelayInSamples = (int)(2 + (fMaxWidthSecs * 2 * fSampleRate));
+
+    for (int i = 0; i < iNumChannels; i++)
+    {
+        for (int j = iDelayInSamples; j < iBlockSize; j++)
+        {
+            EXPECT_NEAR(ppfAudioData[i][j - iDelayInSamples], ppfOutputData[i][j], 1e-3F);
+        }
+    }
+}
+
+// DC input stays DC ouput regardless of parametrization.
+TEST_F(CTestVibrato, DCEqual)
+{
+    for (int i = 0; i < iNumChannels; i++)
+    {
+        CSynthesis::generateDc(ppfAudioData[i], iBlockSize);
+    }
+
+    pVibrato -> reset();
+
+    fModWidth = 0.05;
+
+    pVibrato -> CVibrato::init(fDelay, fSampleRate, fModWidth, fModFreq, iNumChannels);
+
+    testProcess(ppfOutputData, iBlockSize);
+
+    int iDelayInSamples = (int)(2 + (fMaxWidthSecs * 2 * fSampleRate));
+
+    for (int i = 0; i < iNumChannels; i++)
+    {
+        for (int j = iDelayInSamples; j < iBlockSize; j++)
+        {
+            EXPECT_NEAR(ppfAudioData[i][j - iDelayInSamples], ppfOutputData[i][j], 1e-3F);
+        }
+    }
+}
+
+
+
+// Varying input block size.
+TEST_F(CTestVibrato, VaryBlockSize)
+{
+    float** ppfOutBlock  = new float*[iNumChannels];
+    
+    for (int i = 0; i < iNumChannels; i++)
+    {
+        ppfOutBlock[i] = new float[iBlockSize];
+    }
+    
+    pVibrato -> reset();
+    
+    pVibrato -> CVibrato::init(fDelay, fSampleRate, fModWidth, fModFreq, iNumChannels);
+    
+    testProcess(ppfOutputData, iBlockSize);
+    testProcess(ppfOutBlock, 459);
+    
+    for (int i = 0; i < iNumChannels; i++)
+    {
+        CHECK_ARRAY_CLOSE(ppfOutBlock[i], ppfOutputData[i], iBlockSize, 1e-3F);
```

Comment:
I'm a little confused here - `ppfOutBlock` is the same size as `ppfOutputData`, but only the first 459 frames should be filled with meaningful data (because of the `testProcess` call above). So why would `ppfOutBlock[i]` be close to `ppfOutputData[i]` when `i >= 459`?
Also - should the vibrato be reset between the two `testProcess` calls above? Otherwise it seems that the delay line and LFO will not start in the same state in those two cases.